### ISSUE TITLE
Permission error when trying to access a volume defined in an image with custom user and --userns=auto enabled

### ISSUE THREAD

**Issue by elpepino89** [containers/podman#24718]
### Issue Description

When executing a container with `--userns=auto` and using an image including a volume mount (e.g. `VOLUME["/data"]`) and a specific user (e.g. redis with uid 999) I get a permission error when trying to access the volume. The error can be avoided by either dropping `--userns=auto` or by adding `--volume=my-vol:/data:U`. 

It seems that on container startup the ownership of directory `~/.local/share/containers/storage/volumes/[my-vol]/_data` is changed to the container user, if the uid:gid is not part of the current user namespace. This process seems to have an error, because the uid:gid is not calculated correctly: Instead of having an uid:guid from the current user namespace I get the uid:gid without any user namespace. 

### Steps to reproduce the issue

A concrete example just focusing on the uid for simplicity:
```
cat /etc/subuid
podman:1000000:10000000
```

Start a container to reserve some subuids
```
podman run -d --rm --userns=auto:size=100000 ubi8 sleep 1800
```

Start a redis container using uid=999 as redis user and having a VOLUME["/data"] definition.
```
podman run -it --rm --userns=auto:size=100000 --entrypoint=bash docker.io/redis -c " touch /data/test; ls -la /data; sleep 10"
total 12
drwxr-xr-x. 2 nobody nogroup 4096 Nov 12 02:37 .
dr-xr-xr-x. 1 root   root    4096 Nov 30 12:55 ..
touch: cannot touch '/data/test': Permission denied
```

And checking the volume on the host system, it's clear that the _data folder has the wrong ownership, compared to the parent which has the right ownership:
```
ls -la /home/podman/.local/share/containers/storage/volumes/*/_data/
total 8
drwxr-xr-x. 2 1000998 1000998 4096 Nov 12 03:37 .
drwx------. 3 1100000 1100000 4096 Nov 30 13:55 ..
```

The _data folder has the same ownership as when the container was not created with `--userns=auto:size=100000`, which of course cannot work.
```
podman run -it --rm --entrypoint=bash docker.io/redis -c "touch /data/test; ls -la /data; sleep 5"
total 12
drwxr-xr-x. 2 redis redis 4096 Nov 30 12:55 .
dr-xr-xr-x. 1 root  root  4096 Nov 30 12:55 ..
-rw-r--r--. 1 root  root     0 Nov 30 12:55 test
```
```
ls -la /home/podman/.local/share/containers/storage/volumes/*/_data/
total 8
drwxr-xr-x. 2 1000998 1000998 4096 Nov 30 13:55 .
drwx------. 3 podman  podman  4096 Nov 30 13:55 ..
-rw-r--r--. 1 podman  podman     0 Nov 30 13:55 test
```

### Describe the results you received

The uid of the _data folder is 1000998 (nobody) instead of 1100000 (root). The uid will be set correctly if adding `--volume=my-vol:/data:U`, but this comes with a performance impact for big volumes and dropping `--userns=auto:size=100000` comes with a security impact.

### Describe the results you expected

I expect that the _data folder has the same (correct) ownership as his parent. More I expect that the root user of the container is owning the volume and not some "random" uid which is not part of the user namespace of the container.

For my concrete example I expect:
```
ls -la /home/podman/.local/share/containers/storage/volumes/*/_data/
total 8
drwxr-xr-x. 2 1100000 1100000 4096 Nov 30 13:56 .
drwx------. 3 1100000 1100000 4096 Nov 30 13:56 ..
-rw-r--r--. 1 1100000 1100000    0 Nov 30 13:56 test
```

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.37.5
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.12-1.el9.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: eb379dceb7efebd9a9d6b3349a57424d83483065'
  cpuUtilization:
    idlePercent: 99.81
    systemPercent: 0.1
    userPercent: 0.1
  cpus: 8
  databaseBackend: sqlite
  distribution:
    distribution: almalinux
    version: "9.5"
  eventLogger: journald
  freeLocks: 2047
  hostname: twinit-podman
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1111
      size: 1
    - container_id: 1
      host_id: 1000000
      size: 10000000
    uidmap:
    - container_id: 0
      host_id: 1111
      size: 1
    - container_id: 1
      host_id: 1000000
      size: 10000000
  kernel: 5.14.0-503.14.1.el9_5.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 65189359616
  memTotal: 66931347456
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.12.1-1.el9.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.12.1
    package: netavark-1.12.2-1.el9.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.12.2
  ociRuntime:
    name: crun
    package: crun-1.16.1-1.el9.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.16.1
      commit: afa829ca0122bd5e1d67f1f38e6cc348027e3c32
      rundir: /run/user/1111/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20240806.gee36266-2.el9.x86_64
    version: |
      pasta 0^20240806.gee36266-2.el9.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: false
    path: /run/user/1111/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.3.1-1.el9.x86_64
    version: |-
      slirp4netns version 1.3.1
      commit: e5e368c4f5db6ae75c2fce786e31eef9da6bf236
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 34325131264
  swapTotal: 34325131264
  uptime: 1h 38m 9.00s (Approximately 0.04 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/podman/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/podman/.local/share/containers/storage
  graphRootAllocated: 215892250624
  graphRootUsed: 15172530176
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/user/1111/containers
  transientStore: false
  volumePath: /home/podman/.local/share/containers/storage/volumes
version:
  APIVersion: 5.2.2
  Built: 1731426550
  BuiltTime: Tue Nov 12 16:49:10 2024
  GitCommit: ""
  GoVersion: go1.22.7 (Red Hat 1.22.7-2.el9_5)
  Os: linux
  OsArch: linux/amd64
  Version: 5.2.2
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional information

I have the error on alma linux 9 with podman 5.2.2 with selinux enabled. I didn't have the error before on ubuntu 24.04 without selinux enabled and with podman ~4.9.

**Comment by giuseppe** [comment]
it is already fixed upstream with 432325236b7d195045edb9a4cc47be00f71d407f

**Comment by giuseppe** [comment]
duplicate issue: https://github.com/containers/podman/issues/23347
